### PR TITLE
Fixed #27637 -- Fixed timesince, timeuntil in leap year edge case.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [27637](https://code.djangoproject.com/ticket/27637).

### CODE REVIEW

**Comment by claudep** [comment]
Could you add release notes for 1.10, and possibly 1.9? I'd be for pushing the fix in 1.9, too, unless Tim is opposed to that.

**Comment by felixxm** [comment]
I added release notes.

File: django/utils/timesince.py [hunk 1]
```diff
@@ -40,10 +40,15 @@ def timesince(d, now=None, reversed=False):
...
 
-    delta = (d - now) if reversed else (now - d)
+    if reversed:
+        d, now = now, d
+    delta = now - d
 
     # Deal with leapyears by subtracing the number of leapdays
-    delta -= datetime.timedelta(calendar.leapdays(d.year, now.year))
+    leapdays = calendar.leapdays(d.year, now.year)
+    if leapdays != 0 and calendar.isleap(d.year) and d.date() > datetime.date(d.year, 2, 29):
```
Comment (timgraham) [inline django/utils/timesince.py]: I don't see any test failures if `and d.date() > datetime.date(d.year, 2, 29)` is removed. Could you test test that boundary if it's needed?

File: docs/releases/1.10.5.txt [hunk 2]
```diff
@@ -14,3 +14,6 @@ Bugfixes
 
 * Fixed occasional missing plural forms in ``JavaScriptCatalog``
   (:ticket:`27418`).
+
+* Fixed the `timesince` and `timeuntil` filters regression that caused the
```
Comment (timgraham) [inline docs/releases/1.10.5.txt]:
```
* Fixed a regression in the ``timesince`` and ``timeuntil`` filters that caused incorrect results
  for dates after a leap day in a leap year.
```

File: docs/releases/1.9.13.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+===========================
+Django 1.9.13 release notes
+===========================
+
+*Under development*
+
+Django 1.9.13 fixes several bugs in 1.9.12.
```
Comment (timgraham) [inline docs/releases/1.9.13.txt]: several bugs -> a bug
(it's unlikely more bug fixes will be added)

File: django/utils/timesince.py (hunk 1 above)
Comment (felixxm) [inline django/utils/timesince.py]: `timesince` and `timeuntil` returns result up to two adjacent units, hence we can always subtract this one leap day and result will be this same, e.g. `365 -> 1 year`, `366 -> 1 year`. I suppose that we can omit this condition and change release notes to `Fixed a regression in the ``timesince`` and ``timeuntil`` filters that caused incorrect results for dates in a leap year.`

Comment (timgraham) [inline django/utils/timesince.py]: Seems okay to me.

**Comment by felixxm** [comment]
Thanks for review.

**Pull request merged**